Assets/Scripts/Scriptable Objects/ChargeBehaviour.cs

```diff
@@ -0,0 +1,86 @@
+                        break;
+                    }
+                }
+            }
+            // If the agent finishes pathfinding, or the player's position is out of range AND somewhere behind the furniture, stop the charge.
+            Vector3 targetDir = knowledge.PlayerPosition - knowledge.AITransform.position;
+            float angle = Vector3.Angle(targetDir, knowledge.AITransform.forward);
+            if (knowledge.Agent.remainingDistance < 1f || (Vector3.Distance(knowledge.AITransform.position,knowledge.PlayerPosition) > turnRange && angle > 120))
+            {
+                charging = false;
+                knowledge.Agent.SetDestination(knowledge.AITransform.position); // Somehow relying on isStopped doesn't work. This code however, DOES work!
+            }
+        }
+        else
+        {
+            // Make sure the furniture is stationary and always turning to face the player, counting on a timer.
+            knowledge.Agent.isStopped = true;
+            Quaternion lookRotation = Quaternion.LookRotation((new Vector3(knowledge.PlayerPosition.x, knowledge.AITransform.position.y, knowledge.PlayerPosition.z) - knowledge.AITransform.position).normalized);
+            knowledge.AITransform.rotation = Quaternion.Slerp(knowledge.AITransform.rotation, lookRotation, Time.deltaTime * 5);
+            timeSinceLastCharge += Time.deltaTime;
+            // Once time is up, let the agent move again, start the charge and reset some variables.
+            if (timeSinceLastCharge >= chargeDelay)
+            {
+                knowledge.Agent.isStopped = false;
+                charging = true;
+                hitPlayer = false;
+                timeSinceLastCharge = 0;
+                // Get the position to charge towards. This is typically at around where the player is, except slightly further. As a fallback it will try to pathfind directly to the player instead if the navmesh search fails.
+                Vector3 direction = Quaternion.AngleAxis(180, Vector3.up) * (knowledge.PlayerPosition - knowledge.AITransform.position).normalized;
+                Vector3 destination = knowledge.PlayerPosition - (direction * 2f * turnRange);
+                if (NavMesh.SamplePosition(destination, out NavMeshHit hit, turnRange, NavMesh.AllAreas))
+                {
+                    knowledge.Agent.SetDestination(hit.position);
+                }
+                else
+                {
+                    knowledge.Agent.SetDestination(knowledge.PlayerPosition);
+                }
+            }
+        }
+	}
+    public override void Entry(ref Knowledge knowledge)
+    {
+        // Setup the collider and speed
+        hitbox = knowledge.AITransform.GetComponent<Collider>();
+        knowledge.Agent.speed = knowledge.Info.speed * 3;
+    }
+    public override void Exit(ref Knowledge knowledge) { }
```

Comment:
Looks like the speed isn't being reset, meaning the furniture is always at 3x speed after it enters this behaviour?